### ISSUE TITLE
socket: Function not implemented on 4.16

### ISSUE THREAD

**Issue by sishanks** [docker/for-mac#6689]
- [ x] I have tried with the latest version of Docker Desktop
  - [ x] I have tried disabling enabled experimental features
  - [ x] I have uploaded Diagnostics
  - Diagnostics ID: 458A8656-9FBB-4106-AB45-8C9AD55EF97F/20230118094043

### Expected behavior
ability to use i386/debian:10 on intel mac version

### Actual behavior

 (happened after upgrading from 4.15 to 4.16.1) - seems to be 32bit related. Was working on previous version.

### Information

  - macOS Version: 11.7.2
  - Intel chip or Apple chip: intel
  - Docker Desktop Version: 4.16.1

### Steps to reproduce the behavior

Using docker-mac (intel)
```
docker run --platform=linux/386 -it --rm i386/debian:10
root@71dfc1d7ca5b:/# ping 8.8.8.8
ping: socket: Function not implemented
```
using 64bit version works…
```
docker run -it --rm debian:10
root@0a4ef754f5c6:/# ping 8.8.8.8
PING 8.8.8.8 (8.8.8.8) 56(84) bytes of data.
64 bytes from 8.8.8.8: icmp_seq=1 ttl=36 time=25.3 ms
```

**Comment by Jbat1Jumper** [comment]
Running into the same issue here

Can confirm that this also happens in 4.16.0

**Comment by thaJeztah** [comment]
@gabriellavengeo could this be related to `AF_VSOCK` ?

**Comment by gabriellavengeo** [comment]
We identified the problem, it will be fixed in the upcoming patch release (4.16.2).

**Comment by mat007** [comment]
We have just released Docker Desktop 4.16.2 with a fix!

**Comment by Stenzzy** [comment]
> 
> 
> 
> 
>   - [ x] I have tried with the latest version of Docker Desktop
>   - [ x] I have tried disabling enabled experimental features
>   - [ x] I have uploaded Diagnostics
>   - Diagnostics ID: 458A8656-9FBB-4106-AB45-8C9AD55EF97F/20230118094043
> 
> ### Expected behavior
> ability to use i386/debian:10 on intel mac version
> 
> ### Actual behavior
> 
>  (happened after upgrading from 4.15 to 4.16.1) - seems to be 32bit related. Was working on previous version.
> 
> 
> ### Information
> 
>   - macOS Version: 11.7.2
>   - Intel chip or Apple chip: intel
>   - Docker Desktop Version: 4.16.1
> 
> ### Output of `/Applications/Docker.app/Contents/MacOS/com.docker.diagnose check`
> 
> ### Steps to reproduce the behavior
> 
> 
> Using docker-mac (intel)
> ```
> docker run --platform=linux/386 -it --rm i386/debian:10
> root@71dfc1d7ca5b:/# ping 8.8.8.8
> ping: socket: Function not implemented
> ```
> using 64bit version works…
> ```
> docker run -it --rm debian:10
> root@0a4ef754f5c6:/# ping 8.8.8.8
> PING 8.8.8.8 (8.8.8.8) 56(84) bytes of data.
> 64 bytes from 8.8.8.8: icmp_seq=1 ttl=36 time=25.3 ms
> ```
> "X"
